include/plugin.hpp

```diff
@@ -123,18 +129,201 @@ class meminfo_plugin : public scorep::plugin::base<meminfo_plugin,
+            }
+
+            ++line_nr;
+        }
+
+        // CUSTOM
+
+        std::regex regex_custom(regex_custom_str);
+        std::smatch match_custom;
+        std::string s;
+
+        {
+            s = "MemUsed";
+            if (std::regex_match(s, match_custom, regex_custom)) {
+                results.emplace(line_nr, std::pair{s, "B"});
+            }
+            ++line_nr;
+        }
+
+        {
+            s = "SwapUsed";
+            if (std::regex_match(s, match_custom, regex_custom)) {
+                results.emplace(line_nr, std::pair{s, "B"});
+            }
+            ++line_nr;
+        }
+
+        return results;
+    }
+
+    void parse(std::map<std::int64_t, std::vector<std::int64_t>>& data)
+    {
+        std::string line;
+        std::ifstream meminfo("/proc/meminfo");
+
+        int mem_total;
+        int mem_free;
+        int buffers;
+        int cache;
+        int swap_total;
+        int swap_free;
+        int swap_cached;
+
+        std::int64_t line_nr = 0;
+
+        while (std::getline(meminfo, line)) {
+            std::smatch match;
+
+            bool run = false;
+            bool save = false;
+
+            auto handle = data.find(line_nr);
+            if (handle != data.end()) {
+                std::regex_match(line, match, regex_parse);
+
+                std::int64_t value = std::stoll(match[1].str());
+                std::string unit = match[2].str();
+
+                if (unit.size() == 2 && std::tolower(unit[1]) == 'b') {
+                    switch (std::tolower(unit[0])) {
+                    case 't':
+                        value *= 1024;
+                    case 'g':
+                        value *= 1024;
+                    case 'm':
+                        value *= 1024;
+                    case 'k':
+                        value *= 1024;
+                    default:
+                        break;
+                    }
+                }
+
+                if (mem_total_pos == line_nr) {
+                    mem_total = value;
+                }
+                if (mem_free_pos == line_nr) {
+                    mem_free = value;
+                }
+                if (buffers_pos == line_nr) {
+                    buffers = value;
+                }
+                if (cache_pos == line_nr) {
+                    cache = value;
+                }
+                if (swap_total_pos == line_nr) {
+                    swap_total = value;
+                }
+                if (swap_free_pos == line_nr) {
+                    swap_free = value;
+                }
+                if (swap_cached_pos == line_nr) {
+                    swap_cached = value;
+                }
+
+                handle->second.push_back(value);
+            }
+            ++line_nr;
+        }
+
+        // MemUsed
+        auto handle = data.find(mem_used_pos);
+        if (handle != data.end()) {
```

Comment:
Since we are on C++17, you can use the handy if statement initializer (here and in many similar places) https://en.cppreference.com/w/cpp/language/if#If_Statements_with_Initializer

Reply:
changed